docs/getting_started/how_to_build_an_api.md

```diff
@@ -0,0 +1,215 @@
+
+| field       | type     | default     | description     |
+| :------------- | :----------: | :----------: | -----------: |
+| Read |  | 60s | Read, write and idle timeouts. You need to give at least 1s as timeout.\nRead timeout is the maximum duration for reading the entire request, including the body. |
+| Write |  | 60s | Write timeout is the maximum duration before timing out writes of the response. |
+| Idle |  | 60s | Idle timeout is the maximum amount of time to wait for the next request when keep-alives are enabled |
+
+[structmd end]:#
+
+## Practical example: money-exchange application
+
+In `examples/getting_started/api` we can see the _money-exchange_ application. To run it:
+
+1. open a terminal, navigate to `examples/getting_started/api`, and type `$ go run main.go`
+2. open a second terminal, and type `curl localhost:8080/euro/10/GBP` or `curl localhost:8080/euro-at-date/10/USD/2021-01-03T00:00:00Z`
+
+To understand what _money-exchange_ does, let us start by looking at its _main_ function:
+
+[embedmd]:# (../../examples/getting_started/api/main.go /func main/ /\n}/)
+```go
+func main() {
+	app := application.New(
+		application.WithConfigFile("config.dist.yml", "yml"),
+		application.WithKernelSettingsFromConfig,
+		application.WithLoggerHandlersFromConfig)
+
+	app.Add("api", apiserver.New(definer.ApiDefiner))
+	app.Add("currency", currency.NewCurrencyModule())
+
+	app.Run()
+}
+```
+
+This application creates a kernel, using _config.dist.yml_ for its configuration, adds module _"currency"_ and _"api_ to it, then starts the kernel.
+
+These are the contents of _config.dist.yml_:
+```
+env: dev
+
+app_project: gosoline
+app_family: example
+app_name: money-exchange
+
+kernel:
+  killTimeout: 1s
+
+kvstore:
+  currency:
+    type: chain
+    in_memory:
+      max_size: 500000
+    application: money-exchange
+    elements: [inMemory]
+    ttl: 30m
+```
+
+The most interesting config is the currency key value store. This is defined as `inMemory` and serves as a local database.
+
+The _"currency"_ module is already defined by Gosoline, and will use this kvstore to store the exchange rates for various currencies. Its main functionalities are:
+- it makes an initial call to an external endpoint to get exchange rates, and stores them in a kvstore
+- it occasionally makes more calls to obtain exchange rates, and keep the kvstore updates
+
+The _"app"_ module is our api server, notice that it was created with a call to `apiserver.New`:
+
+[embedmd]:# (../../examples/getting_started/api/main.go /app.Add\(\"api\"/ /\)/)
+```go
+app.Add("api", apiserver.New(definer.ApiDefiner)
+```
+
+The code for _ApiDefiner_ is as follows:
+
+[embedmd]:# (../../examples/getting_started/api/definer/definer.go /func ApiDefiner/ /\n}/)
+```go
+func ApiDefiner(ctx context.Context, config cfg.Config, logger log.Logger) (*apiserver.Definitions, error) {
+	definitions := &apiserver.Definitions{}
+
+	currencyService, err := currency.New(ctx, config, logger)
```

Comment:
Why is this not created by the handlers? Why does an Api definer need to create the dependencies of its dependencies?